lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceEvent.java

```diff
@@ -58,4 +60,9 @@
    * @return Runtime name of the resource
    */
   String getRuntimeName();
+
+  /**
+   * @return node label expression of the resource
+   */
+  Optional<Set<String>> getNodeLabelExpressions();
```

Comment:
Also, why do we need to wrap `Set` into `Optional`? An empty set should be good enough if you ask me (both `Set` and `Optional` are monads, after all) :)